src/EditorFeatures/CSharpTest/UseLocalFunction/UseLocalFunctionTests.cs

```diff
@@ -1777,6 +1777,59 @@ void Goo()
 }");
         }
 
+        [WorkItem(23872, "https://github.com/dotnet/roslyn/issues/23872")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        public async Task TestSimpleInitialization_SingleLine4()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void Goo()
+    {
+        Func<int, int[]> [||]onUpdateSolutionCancel = x => { return null; };
```

Comment:
❔ How is this different from `TestSimpleInitialization_SingleLine2`? Or was this test passing before this change was made?

Reply:
So the problem with the previous fix is that it worked for single token return types (i.e. 'void' or 'int') but not multi-token return types.  That's because it would place a space after teh return type, but the formatting engine was checking if there was an elastic trivia after the first *token*.  

So, having ```int[]``` here demonstrated that the old fix didn't work.  But the new fix does work properly here.

Reply:
@CyrusNajmabadi I'm trying that in my branch but `int[]` is working totally fine, are you sure it's caused by it and not async?